test/plugins/test-trace-generic-pool.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+if (!process.env.GCLOUD_PROJECT) {
+  console.log('The GCLOUD_PROJECT environment variable must be set.');
+  process.exit(1);
+}
+
+var assert = require('assert');
+var common = require('./common');
+
+describe('generic-pool-2', function() {
+  var ROOT_SPAN = 'root-span';
+  var CHILD_SPAN_1 = 'child-span-1';
+  var CHILD_SPAN_2 = 'child-span-2';
+
+  var api;
+  var genericPool;
+  before(function() {
+    api = require('../..').start({ samplingRate: 0, stackTraceLimit: 0 });
+    genericPool = require('./fixtures/generic-pool-2');
+  });
+
+  after(function() {
+    common.stopAgent(api);
+  });
+
+  it('perserves context', function(done) {
+    var config = {
+      name: 'generic-pool-2 test',
+      create: function(callback) {
+        callback(function() {
+          var childSpan = api.createChildSpan({ name: CHILD_SPAN_2 });
+          assert.ok(childSpan);
+          childSpan.endSpan();
+        });
+      },
+      destroy: function(fn) {
+      }
+    };
+
+    var pool;
+    api.runInRootSpan({ name: ROOT_SPAN }, function(span) {
+      pool = new genericPool.Pool(config);
+      span.endSpan();
+    });
+
+    pool.acquire(function(err, fn) {
+      assert.ifError(err);
+      var childSpan = api.createChildSpan({ name: CHILD_SPAN_1 });
```

Comment:
I don't think we want to be able to create a child of `ROOT_SPAN` here since we are calling `acquire` from outside the context in which `ROOT_SPAN` is active. The general correctness condition here is that if you call `pool.acquire` from inside a `runInRootSpan` callback, then the same context is available when `pool.acquire` is called and inside the passed to `pool.acquire`. This is analogous to what you did below for generic-pool@3.